src/recipes/heartbeat-by-user-first-impression/index.js

```diff
@@ -241,10 +241,24 @@ let run = function (state, extras) {
 
   let eUrls = [
     `https://qsurvey.mozilla.com/s3/Firefox-USE-Survey?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`,
-    `http://qsurvey.mozilla.com/s3/PBM-Survey-Genpop-41?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`
+    `https://qsurvey.mozilla.com/s3/PBM-Survey-Genpop-41?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`,
+    `https://qsurvey.mozilla.com/s3/Heartbeat-Bright-Spots?source=heartbeat&surveyversion=${VERSION}&updateChannel=${state.updateChannel}&fxVersion=${state.fxVersion}`
   ];
 
-  let engagementUrl = eUrls[~~(Math.random() <= 0.50)];
+  let cutBreaks = function (arr, breaks, rng=Math.random()) {
```

Comment:
This should be in `recipe-utils.js` and I'd definitely add some tests for it. Because you awesomely made rng an argument, it should be pretty easy to test and verify.